loader/loader.go

```diff
@@ -349,6 +350,25 @@ func (l *loader) GetAllSnapshotsForBuildPipelineRun(ctx context.Context, c clien
 	return &snapshots.Items, nil
 }
 
+// GetAllSnapshotsForGivenRepo returns all Snapshots for the associated Pull Request for the same repo url.
+// In the case the List operation fails, an error will be returned.
+// gitops.PipelineAsCodePullRequestAnnotation is also a label
+func (l *loader) GetAllSnapshotsForGivenRepo(ctx context.Context, c client.Client, application *applicationapiv1alpha1.Application, repoUrl string) (*[]applicationapiv1alpha1.Snapshot, error) {
```

Comment:
I still think that this function should be more focused and changed into something like `GetAllSnapshotsForPR` or similar. Basically it would also need to filter on the PR number annotation.

The reason for this is that for a lot of users, this query will return all of their Snapshots, meaning that it would be a more heavy API call, and also we'd have to loop through all of them.

Reply:
Changed, so the function filters only the snapshots with a given PR.